src/lua/LuaSerializer.cpp

```diff
@@ -435,6 +439,58 @@ void LuaSerializer::FromJson(const Json &jsonObj)
 	LUA_DEBUG_END(l, 0);
 }
 
+void LuaSerializer::SaveComponents(Json &jsonObj, Space *space)
+{
+	Json &bodies = jsonObj["space"]["bodies"];
+	if (!bodies.is_array())
+		return;
+
+	// Note: this loop relies on the ordering and contents of Space::m_bodies not changing
```

Comment:
This is a dark comment.

Reply:
Indeed it is. I'd like a more generic solution that serializes Body references similar to how lua table references are serialized, but the lua serialization is a general pile of order-dependent kludge and nothing can solve that. (No offense to the original author of LuaSerializer is meant, it's very brilliant for the domain of the problem it needs to solve and there's really no way around the ordering issues)

Reply:
> This is a dark comment.

Makes me think of this poem:

_I have seen the dark universe yawning,
Where the black planets roll without aim;
Where they roll in their horror unheeded, without knowledge or lustre or name._